services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/BusinessRulesArchiveBuilder.java

```diff
@@ -0,0 +1,104 @@
+package app.coronawarn.server.services.distribution.assembly.component;
+
+import app.coronawarn.server.services.distribution.assembly.structure.Writable;
+import app.coronawarn.server.services.distribution.assembly.structure.WritableOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.dgc.BusinessRule;
+import app.coronawarn.server.services.distribution.dgc.BusinessRule.RuleType;
+import app.coronawarn.server.services.distribution.dgc.BusinessRuleItem;
+import app.coronawarn.server.services.distribution.dgc.DigitalGreenCertificateToCborMapping;
+import app.coronawarn.server.services.distribution.dgc.exception.DigitalCovidCertificateException;
+import app.coronawarn.server.services.distribution.dgc.exception.FetchBusinessRulesException;
+import app.coronawarn.server.services.distribution.dgc.functions.BusinessRuleItemSupplier;
+import app.coronawarn.server.services.distribution.dgc.functions.BusinessRuleSupplier;
+import java.util.List;
+import java.util.Optional;
+import org.apache.logging.log4j.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
Missing java docs for this class :)

Reply:
Hi Felix. I created a test for the builder but I don't know if it's enough. Let's have a discussion on it.